x-pack/plugins/maps/public/connected_components/edit_layer_panel/join_editor/join_editor.tsx

```diff
@@ -85,26 +94,35 @@ export function JoinEditor({ joins, layer, onChange, leftJoinFields, layerDispla
 
   const renderContent = () => {
     const disabledReason = layer.getJoinsDisabledReason();
+    const tooltipMessage = isFeatureEditorOpenForLayer
+      ? i18n.translate('xpack.maps.filterEditor.isJoinsNotApplied', {
+          defaultMessage: 'Term joins are not applied while editing features',
+        })
+      : null;
+
     return disabledReason ? (
       <EuiCallOut color="warning">{disabledReason}</EuiCallOut>
     ) : (
       <Fragment>
         {renderJoins()}
         <EuiSpacer size="s" />
         <EuiTextAlign textAlign="center">
-          <EuiButtonEmpty
-            onClick={addJoin}
-            size="xs"
-            iconType="plusInCircleFilled"
-            aria-label={i18n.translate('xpack.maps.layerPanel.joinEditor.addJoinAriaLabel', {
-              defaultMessage: 'Add join',
-            })}
-          >
-            <FormattedMessage
-              id="xpack.maps.layerPanel.joinEditor.addJoinButtonLabel"
-              defaultMessage="Add join"
-            />
-          </EuiButtonEmpty>
+          <EuiToolTip position="top" content={tooltipMessage}>
```

Comment:
Instead of disabling the "Add join" button with tooltip, set disabledReason reason to 'Term joins are not applied while editing features' and just render disabledReason callout